### PR TITLE
Autocomplete: Use downshift, update styles

### LINKED ISSUE #1611: Replace Turnstone with Downshift
- [ ] Do not dispatch update on text input change
- [ ] Dispatch on blur and on enter
- [ ] Styling similar to what is (turnstone)

### PR DESCRIPTION
**Pull request by sorja**
Resolves #1611 

- [x] Remove turnstone
- [x] Add [downshift](https://github.com/downshift-js/downshift)
- [x] Save changes onBlur or onSelect
- [x] Update styling (following CountrySelect style mainly)
- [x] Selected item shows a small border left
- [x] Hovered items have blue/active background

https://user-images.githubusercontent.com/5508251/195890188-be16d96b-47fa-4feb-91e0-3b38ad51191e.mov

### CODE REVIEW

**Comment by minotogna** [comment]
can we have a video of current implementation ?

File: src/client/components/Autocomplete/Autocomplete.scss [hunk 1]
```diff
@@ -2,22 +2,50 @@
...
-    font-family: $font-family-1;
-    width: 100%;
-    cursor: pointer;
-    box-shadow: $shadow-3;
-    background-color: $ui-accent-light;
-    > * {
-      padding: $spacing-xxs;
+  transition: all .1s ease-out;
+
+  .autocomplete-dropdown {
```
Comment (minotogna) [inline src/client/components/Autocomplete/Autocomplete.scss]: nested css do not need prefix. use prefix if css is not nested (preferred in this case)

File: src/client/components/Autocomplete/Autocomplete.tsx [hunk 2]
```diff
@@ -1,54 +1,103 @@
...
+
+// Issue importing enum without 'type' from Downshift; use enum for only used types
+enum UseComboboxStateChangeTypes {
+  InputBlur = '__input_blur__',
+  ItemClick = '__item_click__',
+}
 
+type Props = {
   value: string
+  onInputValueChange: (changes: UseComboboxStateChange<any>) => void
```
Comment (minotogna) [inline src/client/components/Autocomplete/Autocomplete.tsx]: do not export downshift implementation

File: src/client/components/Autocomplete/Autocomplete.scss (hunk 1 above)
Comment (minotogna) [inline src/client/components/Autocomplete/Autocomplete.scss]: nested css do not need prefix. in this case better not to nest and leave prefix

File: src/client/components/Autocomplete/Autocomplete.tsx [hunk 3]
```diff
@@ -1,54 +1,103 @@
+/* eslint-disable react/jsx-props-no-spreading */
+// Object spreading used by downshift
```
Comment (minotogna) [inline src/client/components/Autocomplete/Autocomplete.tsx]: better to add it to each row where needed. global disabled is not preferred

File: src/client/components/Autocomplete/Autocomplete.tsx [hunk 4]
```diff
@@ -1,54 +1,103 @@
...
+          })}
+      </div>
     </div>
   )
 }
 
 Autocomplete.defaultProps = {
-  name: 'autocomplete',
-  maxItems: 15,
+  name: undefined,
```
Comment (minotogna) [inline src/client/components/Autocomplete/Autocomplete.tsx:108]: if it's optional, is this default undefined value needed?

File: src/client/pages/AssessmentSection/DataTable/Table/Row/RowData/Cell/Taxon/Taxon.tsx [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import React from 'react'
 
 import axios from 'axios'
+import { UseComboboxStateChange } from 'downshift'
```
Comment (minotogna) [inline src/client/pages/AssessmentSection/DataTable/Table/Row/RowData/Cell/Taxon/Taxon.tsx]: do not import downshift anywhere, just in Autocomplete

**Review by minotogna — CHANGES_REQUESTED** [review]

File: src/client/components/Autocomplete/Autocomplete.tsx (hunk 4 above)
Comment (sorja) [reply src/client/components/Autocomplete/Autocomplete.tsx:108]: we get error otherwise that it's missing defaultProps definition

Comment (minotogna) [reply src/client/components/Autocomplete/Autocomplete.tsx:108]: ok!

File: src/client/pages/AssessmentSection/DataTable/Table/Row/RowData/Cell/Taxon/Taxon.tsx [hunk 6]
```diff
@@ -34,17 +37,29 @@ const Taxon: React.FC<PropsCell> = (props: PropsCell) => {
...
   }
 
-  const listbox = {
-    data: (query: string) =>
-      axios
-        .get(`${ApiEndPoint.ExtData.Taxa.search()}?query=${encodeURIComponent(query)}&limit=15`)
-        .then(({ data }) => data),
-    displayField: 'scientificName',
+  const fetchAutocomplete = async (query: string) => {
+    const { data } = await axios.get(`${ApiEndPoint.ExtData.Taxa.search()}?query=${encodeURIComponent(query)}&limit=15`)
```
Comment (minotogna) [inline src/client/pages/AssessmentSection/DataTable/Table/Row/RowData/Cell/Taxon/Taxon.tsx]: you could pass {params} as second argument, to avoid ?..&..

**Review by minotogna — APPROVED** [review]

**Review by rfontanarosa — APPROVED** [review]

**Pull request merged**